cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,130 @@
+				} else if mode.IsDir() {
+					currentPerm.defaultPermissions = defaultPerm.dir
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "d", fullPath, currentPerm, defaultPerm, verbose)
+					FindRecursive(fullPath, defaultPerm, search, verbose, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions returns true if the permissions are correct (false in another case)
+/* currentPermissions [string] - permissions that the file or directory has (i.e. -rwxrwxr-x)
+ * defaultPermissions [string] - permissions that should have (i.e. rwxrwxr-x)
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
+	return strings.Contains(currentPermissions, defaultPermissions)
+}
+
+// printOutput Prints the data in different formats according to the situation
+/* level [string] 														- used to print the output in a hierarchical way
+ * kind [string] 															- "f" if it is file or "d" in case of a directory
+ * fullPath [string] 				  								- file/directory full path (if verbose only use the name)
+ * currentPerm [currentPermissions]						- data to print relate
+ * 	currentPerm.permissions [string] 					- file/directory current permissions
+ * 	currentPerm.defaultPermissions [string] 	- file/directory default permissions
+ * 	currentPerm.owner [string] 								- file/directory current owner
+ * 	currentPerm.group [string] 								- file/directory current group
+ * 	currentPerm.hasCorrectPermissions [bool] 	- true if the permissions are correct
+ * 	currentPerm.hasCorrectOwner [bool] 				- true if the owner is correct
+ * 	currentPerm.hasCorrectGroup [bool] 				- true if the group is correct
+ * defaultPerm [defaultPermissions] 					- data regarding default permissions
+ * 	defaultPerm.file [string]									- default permissions for files
+ * 	defaultPerm.dir [string] 									- default permissions for directories
+ * 	defaultPerm.owner [string] 								- default owner
+ * 	defaultPerm.group [string] 								- default group
+ */
+func printOutput(level int, kind, fullPath string, currentPerm currentPermissions, defaultPerm defaultPermissions, verbose bool) {
+	if verbose {
+		fullPath = filepath.Base(fullPath)
+	}
+	hierarchy := strings.Repeat(" ", level)
+
+	if currentPerm.hasCorrectPermissions && currentPerm.hasCorrectOwner && currentPerm.hasCorrectGroup { // Everything correct
+		if verbose {
+			fmt.Printf("%s(%s) %s %s %s %s\n", hierarchy, kind, fullPath, currentPerm.permissions, currentPerm.owner, currentPerm.group)
+		}
+	} else if currentPerm.hasCorrectPermissions && (!currentPerm.hasCorrectOwner || !currentPerm.hasCorrectGroup) { // Permissions correct, fails owner or group
+		fmt.Printf("\x1b[31;1m%s(%s) %s %s %s %s (expected %s %s)\n\x1b[0m", hierarchy, kind, fullPath, currentPerm.permissions, currentPerm.owner, currentPerm.group, defaultPerm.owner, defaultPerm.group)
```

Comment:
as I review these, I see that the small helper for rendering ansi colors I pasted above would be useful to declutter this.